packages/@aws-cdk/aws-rds/lib/instance.ts

```diff
@@ -1143,6 +1143,12 @@ export class DatabaseInstanceReadReplica extends DatabaseInstanceNew implements
   constructor(scope: Construct, id: string, props: DatabaseInstanceReadReplicaProps) {
     super(scope, id, props);
 
+    if (props.sourceDatabaseInstance.engine
+        && !props.sourceDatabaseInstance.engine.supportsReadReplicaBackups
+        && props.backupRetention) {
+      throw new Error(`Cannot set \`backupRetention\` for a ${props.sourceDatabaseInstance.engine.engineType} read replica. Automatic backups are not supported.`);
```

Comment:
We actually have a function for this - `engineDescription()` (it's in `lib/private/util.ts`).